src/Features/Core/Portable/ExtractMethod/SelectionResult.cs

```diff
@@ -119,59 +121,55 @@ public TExecutableStatementSyntax GetLastStatement()
             return token.GetRequiredAncestor<TExecutableStatementSyntax>();
         }
 
-        public bool CreateAsyncMethod()
-        {
-            _createAsyncMethod ??= CreateAsyncMethodWorker();
-            return _createAsyncMethod.Value;
-
-            bool CreateAsyncMethodWorker()
-            {
-                var firstToken = GetFirstTokenInSelection();
-                var lastToken = GetLastTokenInSelection();
-                var syntaxFacts = SemanticDocument.GetRequiredLanguageService<ISyntaxFactsService>();
-
-                for (var currentToken = firstToken;
-                    currentToken.Span.End < lastToken.SpanStart;
-                    currentToken = currentToken.GetNextToken())
-                {
-                    // [|
-                    //     async () => await ....
-                    // |]
-                    //
-                    // for the case above, even if the selection contains "await", it doesn't belong to the enclosing block
-                    // which extract method is applied to
-                    if (syntaxFacts.IsAwaitKeyword(currentToken)
-                        && !UnderAnonymousOrLocalMethod(currentToken, firstToken, lastToken))
-                    {
-                        return true;
-                    }
-                }
-
-                return false;
-            }
-        }
-
-        public bool ShouldCallConfigureAwaitFalse()
+        /// <summary>
+        /// Checks all of the nodes within the user's selection to see if any of them satisfy the supplied <paramref
+        /// name="predicate"/>. Will not descend into local functions or lambdas.
+        /// </summary>
+        /// <param name="predicate"></param>
+        /// <returns></returns>
+        private bool CheckNodesInSelection(Func<ISyntaxFacts, SyntaxNode, bool> predicate)
```

Comment:
this is the right wayu to write the loop.  descend the nodes.  do not descend into localfunctions/lambdas, and check only those that intersect the span of interest.  Both ContainsAwaitExpression and ContainsConfigureAwait can now jusst defer to this helper.